src/containers/FindFacility.js

```diff
@@ -95,26 +95,36 @@ function FindFacility({ backend, history }) {
     if (!selectedResult) {
       return;
     }
-
-    if (backend.authLoaded && backend.isLoggedIn()) {
-      backend.dropSiteExists(selectedResult).then((exists) => {
-        if (exists) {
+    const facility = hospital_index.index.id_index[selectedResult];
+    backend
+      .dropSiteExists(selectedResult)
+      .then((exists) => {
+        if (!exists) {
+          // create dropsite with corresponding facilty id and info
+          backend.addDropSite({
+            location_id: selectedResult,
+            dropSiteFacilityName: facility.name,
+            dropSiteCity: facility.city,
+            dropSiteState: facility.state,
+            dropSiteZip: facility.zip,
+          });
+        }
+      })
+      .then(() => {
+        if (backend.authLoaded && backend.isLoggedIn()) {
           history.push(
-            routeWithParams(Routes.DROPSITE_ADMIN, { selectedResult }),
+            routeWithParams(Routes.DROPSITE_ADMIN, {
```

Comment:
I guess the assumption here is that the fact that we've hit this code path means that we did find a result (and the add new dropsite flow is triggered from a separate button, yeah?)

Reply:
correct! in the block above this, we create a placeholder dropsite for a facility from the list if it doesn't exist in our db, so by this point there will be a dropsite to direct a user to if they are already logged in. does that make sense?

Reply:
yep!